### PR TITLE
[WIP] Remove comments app

### LINKED ISSUE #2879: Unimplemented ModerationActionManger model in comments app
`readthedocs.comments.models.ModerationActionManager` is unimplemented, and contains a single `current_approvals` method which doesn't return anything.  This is probably a cleanup issue.

### PR DESCRIPTION
**Pull request by stsewd**
PR on top of #3618 (https://github.com/rtfd/readthedocs.org/pull/3618#issuecomment-373084224)

Closes #2879

### CODE REVIEW

File: readthedocs/projects/models.py [hunk 1]
```diff
@@ -136,6 +136,7 @@ class Project(models.Model):
                     'DirectoryHTMLBuilder">More info</a>.'))
 
     # Project features
+    # TODO: remove this?
     allow_comments = models.BooleanField(_('Allow Comments'), default=False)
     comment_moderation = models.BooleanField(
```
Comment (stsewd) [inline readthedocs/projects/models.py]: Are this fields part of the comment app?

`allow_comments` and `comment_moderation`.

I'm going to see if the tests explode if I remove those lines.

Comment (humitos) [reply readthedocs/projects/models.py]: Probably should be removed also. Can you grep the code by `allow_comments` and `comment_moderation` to see where it's used?

Comment (stsewd) [reply readthedocs/projects/models.py]: ## Relevant files

Here is used to choose the sphinx builder
https://github.com/rtfd/readthedocs.org/blob/aefa1b6067f87e2551d8a0e1254ec545b94008d4/readthedocs/doc_builder/backends/sphinx.py#L217-L218

## No relevant files

Just to show on the dj admin
https://github.com/rtfd/readthedocs.org/blob/aefa1b6067f87e2551d8a0e1254ec545b94008d4/readthedocs/projects/admin.py#L105-L106

Test
https://github.com/rtfd/readthedocs.org/blob/aefa1b6067f87e2551d8a0e1254ec545b94008d4/readthedocs/rtd_tests/tests/test_builds.py#L136

Here is just a comment
https://github.com/rtfd/readthedocs.org/blob/aefa1b6067f87e2551d8a0e1254ec545b94008d4/readthedocs/projects/forms.py#L232-L234

On a template to link to the comments section
https://github.com/rtfd/readthedocs.org/blob/aefa1b6067f87e2551d8a0e1254ec545b94008d4/readthedocs/templates/projects/project_edit_base.html#L28-L30

---

For the comment_moderation there is a template that is very related to the comment model. I think that should be removed without problem, but now I'm curious about the sphinx builder.

Comment (stsewd) [reply readthedocs/projects/models.py]: I remove stuff related to `comment_moderation`, and I see that is very related to `allow_comments`,  but I'm have doubts about this

https://github.com/rtfd/readthedocs.org/blob/aefa1b6067f87e2551d8a0e1254ec545b94008d4/readthedocs/doc_builder/backends/sphinx.py#L217-L218

Comment (humitos) [reply readthedocs/projects/models.py]: I think we can safely remove that `readthedocs-comments` builder and all the `-comments` builders.

**Comment by stsewd** [comment]
@humitos I delete the db fields and everything looks good, should I add those migrations here or wait to https://github.com/rtfd/readthedocs.org/pull/3608?

**Comment by humitos** [comment]
> @humitos I delete the db fields and everything looks good, should I add those migrations here or wait to #3608?

I'd say to add them so it's ready to merge. Then we can squash the migration when needed.

**Comment by stsewd** [comment]
Also we want to remove the comments builder from https://github.com/rtfd/readthedocs-sphinx-ext (I'm not familiar with that repo)

**Comment by humitos** [comment]
> Also we want to remove the comments builder from rtfd/readthedocs-sphinx-ext (I'm not familiar with that repo)

I suppose that you can remove the whole `comments` folder, https://github.com/rtfd/readthedocs-sphinx-ext/tree/master/readthedocs_ext/comments

**Comment by ericholscher** [comment]
> I suppose that you can remove the whole comments folder, https://github.com/rtfd/readthedocs-sphinx-ext/tree/master/readthedocs_ext/comments

Yea, I have an initial PR doing this that could be improved (https://github.com/rtfd/readthedocs-sphinx-ext/pull/31)

**Comment by humitos** [comment]
@stsewd would you like to finish Eric's PR in the other repo so we can go forward whith this one also?

**Comment by humitos** [comment]
> @stsewd would you like to finish Eric's PR in the other repo so we can go forward whith this one also?

Linking: https://github.com/rtfd/readthedocs-sphinx-ext/pull/38

**Comment by ericholscher** [comment]
Merging this, and we can figure out if it has further issues during dev today before deploying it.

**Comment by ericholscher** [comment]
🙏 @stsewd

**Comment by stsewd** [comment]
The only thing missing here are the migrations, I think I can add them to #3608

**Pull request merged**